### PR TITLE
Flesh out benchmark.zig with a more varied workload.

### PR DESCRIPTION
**Pull request by jamii**
Many of the fields in the benchmark were fixed, which made many indexes append-only that would not be append-only in reality. Using varied data lets us actually measure compaction perf.

## Pre-merge checklist

Performance:

* [x] Compare `zig benchmark` on linux before and after this pr.
 ``` sh
# main (before reversed ids)
1223 batches in 72.12 s
load offered = 1000000 tx/s
load accepted = 138660 tx/s
batch latency p00 = 5 ms
batch latency p10 = 21 ms
batch latency p20 = 21 ms
batch latency p30 = 21 ms
batch latency p40 = 21 ms
batch latency p50 = 21 ms
batch latency p60 = 21 ms
batch latency p70 = 21 ms
batch latency p80 = 22 ms
batch latency p90 = 26 ms
batch latency p100 = 2747 ms
transfer latency p00 = 5 ms
transfer latency p10 = 2380 ms
transfer latency p20 = 7490 ms
transfer latency p30 = 12955 ms
transfer latency p40 = 18152 ms
transfer latency p50 = 24565 ms
transfer latency p60 = 32031 ms
transfer latency p70 = 38424 ms
transfer latency p80 = 46090 ms
transfer latency p90 = 54525 ms
transfer latency p100 = 62092 ms

# main
1223 batches in 93.93 s
load offered = 1000000 tx/s
load accepted = 106467 tx/s
batch latency p00 = 5 ms
batch latency p10 = 26 ms
batch latency p20 = 26 ms
batch latency p30 = 27 ms
batch latency p40 = 31 ms
batch latency p50 = 31 ms
batch latency p60 = 32 ms
batch latency p70 = 36 ms
batch latency p80 = 41 ms
batch latency p90 = 42 ms
batch latency p100 = 4041 ms
transfer latency p00 = 5 ms
transfer latency p10 = 2551 ms
transfer latency p20 = 8575 ms
transfer latency p30 = 14509 ms
transfer latency p40 = 21299 ms
transfer latency p50 = 30179 ms
transfer latency p60 = 40889 ms
transfer latency p70 = 51164 ms
transfer latency p80 = 62684 ms
transfer latency p90 = 74054 ms
transfer latency p100 = 83893 ms

# vary values
1223 batches in 97.41 s
load offered = 1000000 tx/s
load accepted = 102659 tx/s
batch latency p00 = 5 ms
batch latency p10 = 26 ms
batch latency p20 = 27 ms
batch latency p30 = 27 ms
batch latency p40 = 31 ms
batch latency p50 = 32 ms
batch latency p60 = 32 ms
batch latency p70 = 37 ms
batch latency p80 = 37 ms
batch latency p90 = 42 ms
batch latency p100 = 3242 ms
transfer latency p00 = 5 ms
transfer latency p10 = 2742 ms
transfer latency p20 = 8865 ms
transfer latency p30 = 15273 ms
transfer latency p40 = 22612 ms
transfer latency p50 = 32206 ms
transfer latency p60 = 43763 ms
transfer latency p70 = 54717 ms
transfer latency p80 = 66686 ms
transfer latency p90 = 77643 ms
transfer latency p100 = 87374 ms

# more accounts
1223 batches in 113.64 s
load offered = 1000000 tx/s
load accepted = 88000 tx/s
batch latency p00 = 5 ms
batch latency p10 = 32 ms
batch latency p20 = 32 ms
batch latency p30 = 33 ms
batch latency p40 = 37 ms
batch latency p50 = 37 ms
batch latency p60 = 38 ms
batch latency p70 = 42 ms
batch latency p80 = 47 ms
batch latency p90 = 47 ms
batch latency p100 = 4220 ms
transfer latency p00 = 5 ms
transfer latency p10 = 3372 ms
transfer latency p20 = 10037 ms
transfer latency p30 = 17540 ms
transfer latency p40 = 26471 ms
transfer latency p50 = 37697 ms
transfer latency p60 = 50828 ms
transfer latency p70 = 63352 ms
transfer latency p80 = 77170 ms
transfer latency p90 = 90804 ms
transfer latency p100 = 103609 ms
  ```
OR
* [ ] I am very sure this PR could not affect performance.

### CODE REVIEW

File: src/benchmark.zig [hunk 1]
```diff
@@ -133,12 +89,40 @@ pub fn main() !void {
...
+    // Init accounts.
+    var batch_accounts = try std.ArrayList(tb.Account).initCapacity(allocator, account_count_per_batch);
+    var account_index: usize = 0;
+    while (account_index < account_count) {
+        // Fill batch.
+        while (account_index < account_count and
+            batch_accounts.items.len < account_count_per_batch)
+        {
+            batch_accounts.appendAssumeCapacity(.{
+                .id = account_index + 1,
```
Comment (sentientwaffle) [inline src/benchmark.zig]: Can you do the `bitReverse` trick on account ids too? That will give more realistic `{debit,credit}_account_id` index performance.

**Review by sentientwaffle — COMMENTED** [review]
bors d+

One comment but other than that lgtm.

File: src/benchmark.zig (hunk 1 above)
Comment (jamii) [reply src/benchmark.zig]: Not much change in perf, surprisingly.

**Pull request merged**